numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1103,23 +1103,37 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
 
     unravel_size = PyArray_MultiplyList(dimensions.ptr, dimensions.len);
 
+    dtype = PyArray_DescrFromType(NPY_INTP);
+    if (dtype == NULL) {
+        goto fail;
+    }
+
     if (!PyArray_Check(indices0)) {
         indices = (PyArrayObject*)PyArray_FromAny(indices0,
                                                     NULL, 0, 0, 0, NULL);
         if (indices == NULL) {
             goto fail;
         }
+
+        if (PyArray_SIZE(indices) == 0) {
```

Comment:
Lets say, it is probably the default, but if you want to think about not leaving the error in place, maybe others like it more than me :).